_posts/2018-08-02-Rust-1.28.md

```diff
@@ -0,0 +1,118 @@
+---
+layout: post
+title: "Announcing Rust 1.28"
+author: The Rust Core Team
+---
+
+The Rust team is happy to announce a new version of Rust, 1.28.0. Rust is a
+systems programming language focused on safety, speed, and concurrency.
+
+If you have a previous version of Rust installed via rustup, getting Rust
+1.28.0 is as easy as:
+
+```bash
+$ rustup update stable
+```
+
+If you don't have it already, you can [get `rustup`][install] from the
+appropriate page on our website, and check out the [detailed release notes for
+1.28.0][notes] on GitHub.
+
+[install]: https://www.rust-lang.org/install.html
+[notes]: https://github.com/rust-lang/rust/blob/master/RELEASES.md#version-1280-2018-08-02
+
+## What's in 1.28.0 stable
+
+### Global Allocators
+
+Allocators are the way that programs in Rust obtain memory from the system at
+runtime. Previously, Rust did not allow changing the way memory is obtained,
+which prevented some use cases. With 1.28.0, the `#[global_allocator]` attribute
+is now stable, which allows Rust programs to set their allocator to the system
+allocator, as well as define new allocators by implementing the [`GlobalAlloc`]
+trait.
+
+The standard library provides a handle to the system allocator, which can be
+used as such:
+
+```rust
+use std::alloc::System;
+
+#[global_allocator]
+static GLOBAL: System = System;
+
+fn main() {
+    let mut v = Vec::new();
+    // This will allocate memory using the system allocator.
+    v.push(1);
+}
+```
+
+However, sometimes you want to define a custom allocator for a given application
+domain. This is also relatively easy to do by implementing the `GlobalAlloc`
+trait. You can read more about how to do this in the [documentation].
+
+[`GlobalAlloc`]: https://doc.rust-lang.org/stable/std/alloc/trait.GlobalAlloc.html
+[documentation]: https://doc.rust-lang.org/stable/std/alloc/trait.GlobalAlloc.html
+
+### Improved error message for formatting
+
+Work on diagnostics continues, this time with an emphasis on formatting:
+
+```rust
+format!("{_foo}", _foo = 6usize);
+```
+
+Previously, the error message emitted here was relatively poor:
+
+```
+error: invalid format string: expected `'}'`, found `'_'`
+  |
+2 |     format!("{_foo}", _foo = 6usize);
+  |                     ^^^^^^^^
```

Comment:
The carets are under `"{_foo}"`. https://play.rust-lang.org/?gist=df6fd444250a1fa26df8ecb3b57017b0&version=stable